src/hera.cpp

```diff
@@ -231,15 +236,29 @@ static void evm_destroy(struct evm_instance* instance)
   free(instance);
 }
 
+int evm_set_option(
+  struct evm_instance* instance,
+  char const* name,
+  char const* value
+) {
+  if (strcmp(name, "fallback") == 0) {
+    struct hera_instance* hera = static_cast<struct hera_instance*>(instance);
+    hera->fallback = strcmp(value, "true") == 0;
+    return 1;
+  }
+  return 0;
+}
+
 struct evm_instance* hera_create()
 {
   struct evm_instance init = {
     .abi_version = EVM_ABI_VERSION,
     .destroy = evm_destroy,
     .execute = evm_execute,
-    .set_option = (evm_set_option_fn)NULL
+    .set_option = evm_set_option
   };
-  struct evm_instance* instance = (struct evm_instance*)calloc(1, sizeof(struct evm_instance));
+  // NOTE: allocating `hera_instance` size here
+  struct evm_instance* instance = (struct evm_instance*)calloc(1, sizeof(struct hera_instance));
   if (instance)
     memcpy(instance, &init, sizeof(init));
```

Comment:
@chfast any suggestions to clean up the mess here in an idiomatic way?

Reply:
Yep, move the initialization of evm_instance to constructor of hera_instance. As in https://github.com/ethereum/evmjit/blob/develop/libevmjit/JIT.cpp#L553-L554.